OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventService.cs

```diff
@@ -0,0 +1,105 @@
+{
+    private readonly IEntityRepositorySoftDeleted<Guid, CompetitiveEvent> competitiveEventRepository;
+    private readonly ILogger<CompetitiveEventService> logger;
+    private readonly IStringLocalizer<SharedResource> localizer;
+    private readonly IMapper mapper;
+
+    public CompetitiveEventService(
+        IEntityRepositorySoftDeleted<Guid, CompetitiveEvent> competitiveEventRepository,
+        ILogger<CompetitiveEventService> logger,
+        IStringLocalizer<SharedResource> localizer,
+        IMapper mapper)
+    {
+        this.competitiveEventRepository = competitiveEventRepository ?? throw new ArgumentNullException(nameof(competitiveEventRepository));
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        this.localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
+        this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+    }
+
+    /// <inheritdoc/>
+    public async Task<CompetitiveEventDto> GetById(Guid id)
+    {
+        logger.LogInformation($"Getting CompetitiveEvent by Id started. Looking Id = {id}.");
+
+        var competitiveEvent = await competitiveEventRepository.GetById(id).ConfigureAwait(false);
+
+        if (competitiveEvent == null)
+        {
+            throw new ArgumentOutOfRangeException(
+                nameof(id),
+                localizer["The id cannot be greater than number of table entities."]);
+        }
+
+        logger.LogInformation($"Successfully got a CompetitiveEvent with Id = {id}.");
+
+        return mapper.Map<CompetitiveEventDto>(competitiveEvent);
+    }
+
+    /// <inheritdoc/>
+    public async Task<CompetitiveEventDto> Create(CompetitiveEventDto dto)
+    {
+        logger.LogInformation("CompetitiveEvent creating was started.");
+
+        var competitiveEvent = mapper.Map<CompetitiveEvent>(dto);
+
+        var newCompetitiveEvent = await competitiveEventRepository.Create(competitiveEvent).ConfigureAwait(false);
+
+        logger.LogInformation($"CompetitiveEvent with Id = {newCompetitiveEvent?.Id} created successfully.");
+
+        return mapper.Map<CompetitiveEventDto>(newCompetitiveEvent);
+    }
+
+    /// <inheritdoc/>
+    public async Task<CompetitiveEventDto> Update(CompetitiveEventDto dto)
+    {
+        logger.LogInformation($"Updating CompetitiveEvent with Id = {dto?.Id} started.");
+
+        ArgumentNullException.ThrowIfNull(dto);
+
+        var competitiveEvent = await competitiveEventRepository.GetById(dto.Id).ConfigureAwait(false);
+
+        if (competitiveEvent is null)
+        {
+            var message = $"Updating failed. CompetitiveEvent with Id = {dto.Id} doesn't exist in the system.";
+            logger.LogError(message);
+            throw new DbUpdateConcurrencyException(message);
+        }
+
+        mapper.Map(dto, competitiveEvent);
+        competitiveEvent = await competitiveEventRepository.Update(competitiveEvent).ConfigureAwait(false);
```

Comment:
It would be nice if the structure of the DTO and the base model were the same.

How will the method work if the DTO model is not equal to the model of an entity?
Required fields will throw exceptions and missed fields will be clear?

This is the common approach of Update in our solution.
In the nearest future, we will change this method to some different partial updates.